js-client/src/nix-search.ts

```diff
@@ -0,0 +1,109 @@
+import { LitElement, css, html } from "lit";
+import { customElement, state } from "lit/decorators.js";
+import "@shoelace-style/shoelace/dist/themes/light.css";
+import "@shoelace-style/shoelace/dist/components/button/button.js";
+import "@shoelace-style/shoelace/dist/components/menu/menu.js";
+import "@shoelace-style/shoelace/dist/components/menu-item/menu-item.js";
+import "@shoelace-style/shoelace/dist/components/input/input.js";
+import "@shoelace-style/shoelace/dist/components/spinner/spinner.js";
+import "@shoelace-style/shoelace/dist/components/alert/alert.js";
+import "@shoelace-style/shoelace/dist/components/icon/icon.js";
+
+interface Operation {
+  type: "search" | "graph";
+  data: string[];
+}
+
+type EventInput = Event & { target: HTMLInputElement };
+
+@customElement("nix-search")
+export class NixSearch extends LitElement {
+  worker = new Worker(new URL("./packages", import.meta.url), {
+    type: "module",
+  });
+
+  @state()
+  pkgs: string[] = [];
+
+  @state()
+  loading = true;
+
+  constructor() {
+    super();
+    this.worker.addEventListener("message", (msg: MessageEvent<Operation>) => {
+      const { type, data } = msg.data;
+      if (type === "search") {
+        this.pkgs = data;
+        this.loading = false;
+      }
+    });
+    this.worker.postMessage({ type: "search", data: "" });
```

Comment:
Why is this needed?

Reply:
It's used to make an initial query and display a list of packages. Not really needed, but then the user doesn't see any packages initially

Reply:
Right, thanks. I would add a comment then
```suggestion
    # trigger a request to get an initial list of packages displayed
    this.worker.postMessage({ type: "search", data: "" });
```